src/engine/stack_mmap_XStream.c

```diff
@@ -0,0 +1,87 @@
+/**
+ * (C) Copyright 2016-2021 Intel Corporation.
+ *
+ * SPDX-License-Identifier: BSD-2-Clause-Patent
+ */
+/**
+ * \file
+ *
+ * This file contains the engine/XStream specific part of the
+ * mmap()'ed ULT stack allocation feature.
+ */
+
+#ifdef ULT_MMAP_STACK
+#include <daos/common.h>
+#include "srv_internal.h"
+#include <errno.h>
+#include <string.h>
+
+/* XXX both thresholds may need to be dynamically determined based on the
+ * number of free stack pool (one per-XStream), and max_nb_mmap_stacks (see
+ * below).
+ */
+#define MAX_PERCENT_FREE_STACKS 20
+#define MAX_NUMBER_FREE_STACKS 2000
+
+/* callback to free stack upon ULT exit during stack_key deregister */
+void free_stack(void *arg)
```

Comment:
Should be:
void
free_stack(void *arg)
{
...

same for all functions in this file.